src/Home.js

```diff
@@ -113,39 +141,39 @@ class Home extends React.Component {
       }
     }
     for (var i = 0; i < selectedModuleIds.length; i++) {
-        for (var j=0; j < courseInfo.length; j++){
-          if (selectedModuleIds[i]===courseInfo[j].related_module_id)
-          selectedCourseData.push(courseInfo[j]);
-        }
+      for (var j=0; j < courseInfo.length; j++){
+        if (selectedModuleIds[i]===courseInfo[j].related_module_id)
+        selectedCourseData.push(courseInfo[j]);
       }
+    }
     return selectedCourseData;
   };
 
   retrieveSelectedModuleTitel(){
-     var userModules = this.state.userModules;
-     var modules = moduleplan.degree_course.modules;
-     var selectedModuleIds= [];
-     var selectedModuleTitels= [];
-     for (var i = 0; i < userModules.length; i++) {
+    var userModules = this.state.userModules;
+    var modules = moduleplan.degree_course.modules;
+    var selectedModuleIds= [];
+    var selectedModuleTitels= [];
+    for (var i = 0; i < userModules.length; i++) {
       if (userModules[i].selected){
         selectedModuleIds.push(userModules[i].module_id);
       }
     }
     for (var i = 0; i < selectedModuleIds.length; i++) {
-        for (var j=0; j < modules.length; j++){
-          if (selectedModuleIds[i]===modules[j].id)
-          selectedModuleTitels.push(modules[j].title);
-        }
+      for (var j=0; j < modules.length; j++){
+        if (selectedModuleIds[i]===modules[j].id)
+        selectedModuleTitels.push(modules[j].title);
       }
-      return selectedModuleTitels;
+    }
+    return selectedModuleTitels;
   };
 
   combineSelectedTitelsAndData(){
     var courseInformation= this.retrieveSelectedCourseInfo();
     var modules = moduleplan.degree_course.modules;
-     for (var i = 0; i < modules.length; i++) {
-        for (var j=0; j < courseInformation.length; j++)
-          if (modules[i].id == courseInformation[j].related_module_id){
+    for (var i = 0; i < modules.length; i++) {
+      for (var j=0; j < courseInformation.length; j++)
+      if (modules[i].id == courseInformation[j].related_module_id){
```

Comment:
Here you should use three equal signs for comparison to prevent automatic type coercion.